### PR TITLE
chore: simplify triggering the E2E tests for Beats

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It creates a Jenkins stage right after the package one, in which we will trigger only one job for the E2E tests. To set the test suites to run, we will use the existing BEATS matrix to configure a collection of suites (if/when needed).

We will convert the collection to a Set to remove duplicates, and willl trigger the E2E job with that configuration.

## Why is it important?
With previous approach, it triggers up to 4 builds for almost the same test suites, and the E2E job supports passing multiple suites as an comma-separated list of strings. As a direct consequence, only one GH check will be added to the PR, insteadd of multiple, one per modified Beat.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [x] Groovy file validated with APM Pipeline validator

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 @Library('apm@current') _
 
+import groovy.transform.Field
+
+/**
+ This is required to store the build status for the downstream jobs.
```
Comment (mdelapenya) [inline .ci/packaging.groovy:8]:
```suggestion
 This is required to store the test suites we will use to trigger the E2E tests.
```

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -285,7 +311,7 @@ def triggerE2ETests(String suite) {
     wait: false
   )
 
-  def notifyContext = "${env.GITHUB_CHECK_E2E_TESTS_NAME} for ${env.BEATS_FOLDER}"
+  def notifyContext = "${env.GITHUB_CHECK_E2E_TESTS_NAME}"
```
Comment (mdelapenya) [inline .ci/packaging.groovy:314]: Only one GH check per PR

File: .ci/packaging.groovy [hunk 3]
```diff
@@ -257,8 +267,24 @@ def release(){
   }
 }
 
+def runE2ETests(){
+  def suites = '' // empty value represents all suites in the E2E tests
+
+  catchError(buildResult: 'UNSTABLE', message: 'Unable to run e2e tests', stageResult: 'FAILURE') {
+    def suitesSet = e2eTestSuites.toSet()
```
Comment (mdelapenya) [inline .ci/packaging.groovy:274]: `toSet()` is allowed by Jenkins sandbox:
- https://github.com/jenkinsci/script-security-plugin/blob/master/src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist

File: .ci/packaging.groovy [hunk 4]
```diff
@@ -257,8 +267,24 @@ def release(){
...
+
+    if (!suitesSet.contains('ALL')) {
+      suitesSet.each { suite ->
+        suites += "${suite},"
```
Comment (mdelapenya) [inline .ci/packaging.groovy:278]: Verified that last comma does not cause any new suite with empty value. See https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-mbp/job/master/512/parameters/

**Review by cachedout — APPROVED** [review]

**Comment by mdelapenya** [comment]
Ok, only one check for the E2E is added to the PR. I'd say it's ready to be merged

**Pull request merged**